src/parcsr_mv/par_csr_communication.c

```diff
@@ -60,11 +64,46 @@ hypre_ParCSRPersistentCommHandleCreate( HYPRE_Int job, hypre_ParCSRCommPkg *comm
    HYPRE_Int num_recvs = hypre_ParCSRCommPkgNumRecvs(comm_pkg);
    MPI_Comm  comm      = hypre_ParCSRCommPkgComm(comm_pkg);
 
+
+#ifdef HYPRE_USING_NODE_AWARE_MPI
+   HYPRE_Int num_requests;
+   hypre_MPI_Request *requests;
+
+   HYPRE_Int *send_sizes;
+   HYPRE_Int *recv_sizes;
+   HYPRE_BigInt num_send_elmts;
+   HYPRE_BigInt num_recv_elmts;
+   MPIX_Request *Xrequest;
+
+   HYPRE_Int node_aware_on = hypre_HandleUsingNodeAwareMPI(hypre_handle());
+   if (comm_pkg->use_neighbor && node_aware_on) {
+      if (comm_pkg->neighbor_comm == NULL) {
+         hypre_printf("Trying to communicate with a NULL communicator\n");
```

Comment:
Please, update error handling. Check here for an example: https://github.com/hypre-space/hypre/blob/36ab29ba0977294a741987c4f921fe02840dff59/src/parcsr_mv/par_csr_matrix.c#L712-L713